### ISSUE TITLE
Edge case bug when using handlebars template 'getUrlField

### ISSUE THREAD

**Issue by jcoolsen** [microsoft-search/pnp-modern-search#437]
Using version 3.14.2.0

Using the templating and the handlebars template getUrlField, it does not handle commas properly. Since the URL and the text are separated by one comma, any comma present in the actual URL or Title is represented by double comma - this is a SharePoint thing.

The getUrlField helper should handle this properly.

Currently, if I have this hypothetical construction:

URL: https://www.google.com/,s
Text: Google alt text

then the encoded version looks like this:
https://www.google.com/,,s, Google alt text

This expression {{getUrlField FullLinkOWSURLH "URL"}} currently produces
https://www.google.com/
while it should have produced
https://www.google.com/,s

This expression {{getUrlField FullLinkOWSURLH "Title"}} currently produces
,s, Google alt text
while it should have produced
Google alt text

Chrome browser Version 84.0.4147.135

**Comment by wobba** [comment]
How do you propose to fix it? What if the label has commas? Hard to figure out where to split.

**Comment by jcoolsen** [comment]
The split has to be where there is a singular comma. Pairs of commas should be converted to single comma after the split. I have some code used in another web part, that I can share tomorrow. If the first set of commas is a triplet, it would be unclear where to split, true. I can do a test to clarify this as well.

**Comment by jcoolsen** [comment]
This regular expression can do the split, leaving the label/text part optional:
`^((?:,,|[^,])+)(?:, (.*))?$`
By testing further, I have determined that for only the URL part, all instances of `,,` (double comma) must be replaced with `,` (single comma) after the split is made, The label part can be used as is. Also, the split is actually done with `, ` (one comma and one whitespace), which makes ambiguites fewer - maybe.

**Comment by wobba** [comment]
@jcoolsen want to try to PR this against the dev branch? 🙂

**Comment by jcoolsen** [comment]
@wobba , yes, I'll help test it, but currently I don't see any changes related to this on the "develop" branch. I'm pretty new to Git so I may be looking at the wrong place. Also shouldn't there be a link from this issue to the change?

**Comment by wobba** [comment]
@jcoolsen You have to fork the dev branch, then create a new branch in your fork. Do the fix, commit, push, and then create a PR against the develop branch in this repo. Should be enough guides out there as this is a general git thing :) If you still don't manage I'll try to help out.

**Comment by wobba** [comment]
v3 no longer supported
